packages/workspace/src/command-line/shared.ts

```diff
@@ -399,44 +412,60 @@ export const getAffected = (affectedNamesFetcher: AffectedFetcher) => (
   return affectedNamesFetcher(sortedProjects, dependencies, tp);
 };
 
-export function getAffectedProjectsWithTarget(target: string) {
-  return getAffected(affectedProjectNamesWithTarget(target));
+export function getAffectedProjectsWithTargetAndConfiguration(
+  target: string,
+  configuration?: string
+) {
+  return getAffected(
+    affectedProjectWithTargetAndConfiguration(target, configuration)
+  );
 }
-export const getAffectedApps = getAffected(affectedAppNames);
-export const getAffectedProjects = getAffected(affectedProjectNames);
-export const getAffectedLibs = getAffected(affectedLibNames);
+export const getAffectedApps = getAffected(affectedApps);
+export const getAffectedProjects = getAffected(affectedProjects);
+export const getAffectedLibs = getAffected(affectedLibs);
 
-export function getAllAppNames() {
-  return getProjectNames(p => p.type === ProjectType.app);
+export function getAllApps() {
+  return getAllProjects(p => p.type === ProjectType.app);
 }
 
-export function getAllLibNames() {
-  return getProjectNames(p => p.type === ProjectType.lib);
+export function getAllLibs() {
+  return getAllProjects(p => p.type === ProjectType.lib);
 }
 
-export function getAllProjectNamesWithTarget(target: string) {
-  return getProjectNames(p => p.architect[target]);
+export function getAllProjectsWithTargetAndConfiguration(
+  target: string,
+  configuration?: string
+): AffectedProject[] {
+  return getAllProjects(
+    proj =>
+      proj.architect[target] &&
+      proj.architect[target].configurations &&
+      proj.architect[target].configurations[configuration]
```

Comment:
this seems weird. it looks like configuration is optional but in this if statement it's used as if it wasn't